File: projects/settings_app/examples/no_headerbar_demo.c
// Example: GTK4 window without traditional headerbar in C

#include <gtk/gtk.h>
#include <adwaita.h>

static void on_close_clicked(GtkWidget *widget, gpointer data) {
    GtkWindow *window = GTK_WINDOW(data);
    gtk_window_destroy(window);
}

static void activate(GtkApplication *app, gpointer user_data) {
    GtkWidget *window;
    GtkWidget *main_box;
    GtkWidget *control_bar;
    GtkWidget *close_btn;
    GtkWidget *title_label;
    GtkWidget *content;

    // Create window without decorations
    window = gtk_application_window_new(app);
    gtk_window_set_title(GTK_WINDOW(window), "Settings");
    gtk_window_set_default_size(GTK_WINDOW(window), 800, 600);

    // Remove window decorations (title bar, borders)
    gtk_window_set_decorated(GTK_WINDOW(window), FALSE);

    // Create main vertical box
    main_box = gtk_box_new(GTK_ORIENTATION_VERTICAL, 0);
    gtk_window_set_child(GTK_WINDOW(window), main_box);

    // Create custom control bar (since we removed decorations)
    control_bar = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 6);
    gtk_widget_set_margin_top(control_bar, 6);
    gtk_widget_set_margin_start(control_bar, 12);
    gtk_widget_set_margin_end(control_bar, 12);
    gtk_box_append(GTK_BOX(main_box), control_bar);

    // Add title label
    title_label = gtk_label_new("Settings");
    gtk_widget_set_hexpand(title_label, TRUE);
    gtk_widget_set_halign(title_label, GTK_ALIGN_START);
    gtk_box_append(GTK_BOX(control_bar), title_label);

    // Add close button
    close_btn = gtk_button_new_with_label("×");
    g_signal_connect(close_btn, "clicked", G_CALLBACK(on_close_clicked), window);
    gtk_box_append(GTK_BOX(control_bar), close_btn);

    // Add main content area
    content = gtk_label_new("Main Settings Content Area");
    gtk_widget_set_vexpand(content, TRUE);
    gtk_box_append(GTK_BOX(main_box), content);

    // Show window
    gtk_window_present(GTK_WINDOW(window));
}

int main(int argc, char **argv) {
    AdwApplication *app;
    int status;

    // Initialize Adwaita
    app = adw_application_new("com.example.NoHeaderbarDemo", G_APPLICATION_DEFAULT_FLAGS);
    g_signal_connect(app, "activate", G_CALLBACK(activate), NULL);

    status = g_application_run(G_APPLICATION(app), argc, argv);
    g_object_unref(app);

    return status;
}

// Compile with:
// gcc -o demo no_headerbar_demo.c `pkg-config --cflags --libs gtk4 libadwaita-1`

File: projects/settings_app/examples/no_headerbar_demo.rs
// Example: GTK4 window without traditional headerbar in Rust

use gtk4::prelude::*;
use gtk4::{Application, ApplicationWindow, Box, Button, Label, Orientation};
use libadwaita as adw;
use libadwaita::prelude::*;

fn main() {
    let app = adw::Application::builder()
        .application_id("com.example.NoHeaderbarDemo")
        .build();

    app.connect_activate(build_ui);
    app.run();
}

fn build_ui(app: &adw::Application) {
    // Method 1: Completely borderless window (no decorations)
    let window = ApplicationWindow::builder()
        .application(app)
        .default_width(800)
        .default_height(600)
        .decorated(false)  // This removes ALL window decorations
        .build();

    // Method 2: Custom minimal header (if you want SOME controls)
    // Uncomment below and comment out .decorated(false) above
    /*
    let window = ApplicationWindow::builder()
        .application(app)
        .default_width(800)
        .default_height(600)
        .build();

    // Create custom minimal header
    let header = gtk4::HeaderBar::builder()
        .show_title_buttons(false)  // No min/max/close buttons
        .title_widget(&gtk4::Box::new(Orientation::Horizontal, 0))  // Empty title
        .build();

    window.set_titlebar(Some(&header));
    */

    // Method 3: Using Libadwaita's Window with custom content
    // This gives you more control over the window appearance
    /*
    let window = adw::Window::builder()
        .application(app)
        .default_width(800)
        .default_height(600)
        .build();

    // Remove the default header bar
    window.set_decorated(false);
    */

    // Create main content
    let main_box = Box::new(Orientation::Vertical, 0);

    // Add custom window controls if needed (for borderless window)
    if !window.is_decorated() {
        let control_bar = Box::new(Orientation::Horizontal, 6);
        control_bar.set_margin_top(6);
        control_bar.set_margin_start(12);
        control_bar.set_margin_end(12);

        // Add custom close button
        let close_btn = Button::with_label("×");
        close_btn.set_valign(gtk4::Align::Center);
        close_btn.connect_clicked(move |_| {
            std::process::exit(0);
        });

        control_bar.append(&Label::new(Some("Settings")));
        control_bar.set_hexpand(true);
        control_bar.append(&close_btn);

        main_box.append(&control_bar);
    }

    // Add main content area
    let content = Label::new(Some("Main Settings Content Area"));
    content.set_vexpand(true);
    main_box.append(&content);

    window.set_child(Some(&main_box));
    window.present();
}

// For Hyprland specifically, you can also use window rules to remove decorations:
// In hyprland.conf:
// windowrulev2 = noborder, class:^(settings-app)$
// windowrulev2 = noshadow, class:^(settings-app)$